### PR TITLE
Switch to `pnpm` and refactor the project accordingly

### PR DESCRIPTION
**Pull request by YacineSteeve**
## What does this PR do ?

* Change `npm` to `pnpm` for packages management
* Improve the CI tasks workflow

### CODE REVIEW

**Comment by YacineSteeve** [comment]
Current dependencies on/for this PR:
* main
  * **PR #96** <a href="https://app.graphite.dev/github/pr/Frelya/react-site-editor/96" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Frelya/react-site-editor/96?utm_source=stack-comment).

File: README.md [hunk 1]
```diff
@@ -16,32 +16,32 @@ development and testing purposes.
...
 ```bash
-npm i
+pnpm i
 ```
 
 4. Start the development server:
 
 ```bash
-npm run dev
+pnpm dev
```
Comment (epoundor) [inline README.md:44]: There's something missing. Not?

File: package.json [hunk 2]
```diff
@@ -5,35 +5,37 @@
...
+        "ui:scripts:init": "pnpm --filter ui scripts:init",
+        "ui:start": "pnpm --filter ui start",
+        "lint": "npx eslint --cache --ext .js,.ts,.jsx,.tsx,.json .",
+        "lint:fix": "pnpm lint --fix",
+        "format": "npx prettier --cache --write ./ '!**/*.{js,ts,jsx,tsx,html,json}'",
+        "ci:style": "pnpm lint:fix && pnpm format",
+        "ci:all": "pnpm ci:style && pnpm test",
+        "push": "pnpm ci:all && git commit -am 'ci: lint and format' && git push"
     },
-    "workspaces": [
```
Comment (epoundor) [inline package.json:32]: Why did U remove the workspaces?

**Review by epoundor — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (YacineSteeve) [reply package.json:32]: With `pnpm`, workspaces are not defined in the `package.json` but in a `pnpm-workspace.yaml` file instead.

File: README.md (hunk 1 above)
Comment (YacineSteeve) [reply README.md:44]: No, nothing is missing. What are you thinking about?

Comment (epoundor) [reply README.md:44]: Something like pnpm run dev

File: README.md [hunk 3]
```diff
@@ -51,15 +51,20 @@ The page builder should now be running at `http://localhost:3000`.
 To deploy the page builder to a production environment, run the following command:
 
 ```bash
-npm run build
+pnpm run build
```
Comment (epoundor) [inline README.md:54]: pnpm run ???

File: README.md (hunk 1 above)
Comment (YacineSteeve) [reply README.md:44]: The `run` option is not mandatory in `pnpm`, as in `yarn`

File: README.md (hunk 3 above)
Comment (YacineSteeve) [reply README.md:54]: Ow, my bad. I forgot to remove it here too

Comment (YacineSteeve) [reply README.md]: Fixed

**Review by epoundor — APPROVED** [review]

**Pull request merged**